library/common/network/configurator.cc

```diff
@@ -60,21 +61,46 @@ namespace {
 #define SUPPORTS_GETIFADDRS
 #endif
 
-std::atomic<envoy_network_t> MobileUtility::preferred_network_{ENVOY_NET_GENERIC};
+SINGLETON_MANAGER_REGISTRATION(network_configurator);
 
-void MobileUtility::setPreferredNetwork(envoy_network_t network) { preferred_network_ = network; }
+constexpr absl::string_view BaseDnsCache = "base_dns_cache";
 
-envoy_network_t MobileUtility::getPreferredNetwork() { return preferred_network_.load(); }
+std::atomic<envoy_network_t> Configurator::preferred_network_{ENVOY_NET_GENERIC};
 
-std::vector<std::string> MobileUtility::enumerateV4Interfaces() {
+envoy_network_t Configurator::setPreferredNetwork(envoy_network_t network) {
+  ENVOY_LOG_EVENT(debug, "network_configuration_network_change", std::to_string(network));
+  return preferred_network_.exchange(network);
+}
+
+envoy_network_t Configurator::getPreferredNetwork() { return preferred_network_.load(); }
+
+void Configurator::refreshDns(envoy_network_t network) {
+  // refreshDns is intended to be queued on Envoy's event loop, whereas preferred_network_ is
+  // updated synchronously. In the event that multiple refreshes become queued on the event loop,
+  // this avoids triggering a refresh for a non-current network.
+  // Note this does NOT completely prevent parallel refreshes from being triggered in multiple
+  // flip-flop scenarios.
+  if (network != preferred_network_.load()) {
+    return;
+  }
+  // TODO(goaway): track event here or are there existing signals we can use?
```

Comment:
This seems to have been done with the below events? Or is this talking about something else?

Reply:
Yep the comment is out of date; thanks.